### PR TITLE
UCP/WIREUP: fix pending counter in case of error

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
@alinask @yosefe pls take a look

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6734/ for details (Mellanox internal link).

File: src/ucp/core/ucp_ep.c [hunk 1]
```diff
@@ -480,7 +480,12 @@ void ucp_ep_err_pending_purge(uct_pending_req_t *self, void *arg)
 {
     ucp_request_t *req      = ucs_container_of(self, ucp_request_t, send.uct);
     ucs_status_t  status    = UCS_PTR_STATUS(arg);
+    ucp_ep_h      ep;
 
+    ep = req->send.ep;
+    if (ucp_wireup_ep_test(ucp_ep_get_wireup_uct_ep(ep))) {
```
Comment (yosefe) [inline src/ucp/core/ucp_ep.c]: can this be part of ucp_wireup_ep_pending_purge?

**Comment by alinask** [comment]
:+1: tested in my err_handle scenario in gtest.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6740/ for details (Mellanox internal link).

File: src/ucp/wireup/wireup_ep.c [hunk 2]
```diff
@@ -195,6 +195,7 @@ ucp_wireup_ep_pending_purge(uct_ep_h uct_ep, uct_pending_purge_callback_t cb,
 
     ucs_queue_for_each_extract(req, &wireup_ep->pending_q, priv, 1) {
         ucp_req = ucs_container_of(req, ucp_request_t, send.uct);
+        --ucp_req->send.ep->worker->wireup_pend_count;
```
Comment (yosefe) [inline src/ucp/wireup/wireup_ep.c]: i think we need to put this inside UCS_ASYNC_BLOCK/UCS_ASYNC_UNBLOCK

**Pull request merged**